extensions/namespace-lookup/src/test/java/io/druid/server/namespace/cache/JDBCExtractionNamespaceTest.java

```diff
@@ -285,24 +326,46 @@ public void testFindNew()
         keyName,
         valName,
         tsColumn,
-        new Period(1)
+        new Period(10)
     );
     extractionCacheManager.schedule(extractionNamespace);
-    while (!fnCache.containsKey(extractionNamespace.getNamespace())) {
-      Thread.sleep(1);
-    }
-    Function<String, String> extractionFn = fnCache.get(extractionNamespace.getNamespace());
+
+    waitForUpdates(1_000L, 2L);
+
     Assert.assertEquals(
+        "sanity check not correct",
         "bar",
-        extractionFn.apply("foo")
+        fnCache.get(extractionNamespace.getNamespace()).apply("foo")
     );
+    return extractionNamespace;
+  }
 
-    insertValues("foo", "baz", "2900-01-01 00:00:00");
-    Thread.sleep(100);
-    extractionFn = fnCache.get(extractionNamespace.getNamespace());
+  private void waitForUpdates(long timeout, long numUpdates) throws InterruptedException
+  {
+    long startTime = System.currentTimeMillis();
+    long pre = 0L;
+    synchronized (updateLock) {
+      pre = updates.get();
+    }
+    do {
+      // Sleep to spare a few cpu cycles
+      Thread.sleep(5);
+      log.info("Waiting for updateLock");
+      synchronized (updateLock) {
+        Assert.assertTrue("Failed waiting for update", System.currentTimeMillis() - startTime < timeout);
```

Comment:
Just a gatekeeper between sleep and `updates.get()`. It was to prevent an annoying race condition in part of the count updating, but that condition may not be present in how the current code works.